### PR TITLE
Upgrade to MQTT Paho 1.2.2

### PR DESCRIPTION
**Pull request by garyrussell**
- fix mock tests for internal client changes
- reduce stop wait for completion time

**cherry-pick to 5.2.x**

### CODE REVIEW

File: spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java [hunk 1]
```diff
@@ -454,9 +458,10 @@ public void testSubscribeFailure() throws Exception {
 			fail("Expected InvocationTargetException");
 		}
 		catch (InvocationTargetException e) {
+			e.printStackTrace();
```
Comment (artembilan) [inline spring-integration-mqtt/src/test/java/org/springframework/integration/mqtt/MqttAdapterTests.java]: I think this one has to go...

**Review by artembilan — CHANGES_REQUESTED** [review]
One nit-pick

**Comment by artembilan** [comment]
... and cherry-picked to `5.2.x`

**Comment by garyrussell** [comment]
and to 5.1.x as 4174bfd6cabcc0e51e058c82061218d4ce49caff

**Pull request merged**